### DISCUSSION TITLE
AB Scripting - Retrieving Task Form field value

### DISCUSSION THREAD

**Discussion by longj08** [sadieblue/agilityblue.workspace-scripts#14] [Q&A] [answered]
We are working on a script to retrieve values from individual fields within a section of a task form. However, when we run the script to get the value from a given field based on a filter, it returns all the fields within the task form.
Is there an example of how we can achieve this? In this instance we just need the value form one of the fields of the task form.

Below are some examples that Jack Long (cc in email) has tried to test the process:

Example 1: 
$form = $task.Forms.sections.Fields.FieldID 
if($form -eq 2056){ 
#$value = $form.ValueAsBoolean 
#$label = $form.Label 
write-output "Yes" 
Write-output "Form: $form" 
write-output "value: $value"
}

Example 2: 
$fieldId = 2056
$task.Forms.sections.fields | ForEach-Object {
#Write-Output $task.forms.sections.fields 
$task.forms.sections.fields | ConvertTo-JSON -Depth 10 
#exit 
$taskInProject = Get-Task $_.fieldId -IncludeForms 
# $taskInProject | ConvertTo-JSON -Depth 10
if ($task.forms.sections.fields | Where-Object { $_.fieldID -eq $fieldId}) { 
write-output "Yes" 
Write-Output "fieldId: " 
Write-Output $task.forms.sections.fields.fieldId 
#Write-Output "TaskFormID: $fieldId" 
#$data = $task.forms.sections.fields.ValueAsBoolean 
#write-output "data: $data" 
#$Id = $task.forms.sections.fields.fieldId 
#write-output "ID: $Id"
return 
}
}

**Comment by mrabb24** [comment, marked as answer]
Hi there,

It looks like you're running into the PowerShell functionality that will select multiple objects back to you as an array when you apply dot-notation syntax on arrays. The "Forms", "Sections", and "Fields" properties are all arrays, so you should iterate through each of them instead of trying to treat them as individual objects.

See if this works better for you:

```powershell
$task.Forms | ForEach-Object {
  $form = $_

  $form.Sections | ForEach-Object {
    $section = $_
 
    # You can now apply a filter on the Fields array if you like:
    $myField = $section.Fields | Where-Object { $_.Label -eq "My Field" }

    # Or, you can access each field by iterating them:
    $section.Fields | ForEach-Object {
      $field = $_
      
      if ($field.Label -eq "My Field") {
        # do stuff
      }
    }
  }
}
```

Let me know if this achieved your desired output,
Matt

**Reply by longj08** [reply]
It works.  Thank you so much, Matt.
